modules/@apostrophecms/widget-type/ui/apos/components/AposWidgetEditor.vue

```diff
@@ -190,11 +195,14 @@ export default {
         ...defaults,
         ...this.modelValue
       };
-      return;
+    } else {
+      this.original = klona(defaults);
+      this.docFields.data = defaults;
     }
-
-    this.original = klona(defaults);
-    this.docFields.data = defaults;
+    if (!this.id) {
+      this.newId = createId();
```

Comment:
Good practice to instantiate all data inside `data` block even with a null value or whatever.
Something like that would be clear, and we would use only newId:
```javascript
return {
   ...
    newId: this.id || createId()
}
```

Reply:
data is reactive, newId does not need to be reactive, so this would be unnecessary performance overhead.

Reply:
(Because of the avoidable overhead, I don't agree it is a best practice to use data for things that are not reactive, although I would not reject a PR for doing it.)

Reply:
I'm not sure this is the kind of performance overhead we should worry about.
I'm saying bad practice because I think it's a non document behavior so could lead to unexpected behavior, also think it's a bit less readable but I'm ok.